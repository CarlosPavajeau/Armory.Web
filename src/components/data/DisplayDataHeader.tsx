import { MouseEventHandler, ReactElement, useEffect } from 'react';
import { withStyles, WithStyles } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import TextField from '@material-ui/core/TextField';
import Toolbar from '@material-ui/core/Toolbar';
import Tooltip from '@material-ui/core/Tooltip';
import Grid from '@material-ui/core/Grid';
import SearchIcon from '@material-ui/icons/Search';
import RefreshIcon from '@material-ui/icons/Refresh';
import { displayData } from '../../common/styles';

export interface DisplayDataHeaderProps extends WithStyles<typeof displayData> {
  handleRefresh: MouseEventHandler<HTMLButtonElement>;
  placeholder: string;
}

const DisplayDataHeader = (props: DisplayDataHeaderProps): ReactElement => {
  const { classes, handleRefresh, placeholder } = props;

  return (
    <AppBar
      position="static"
      color="default"
      elevation={0}
      className={classes.searchBar}
    >
      <Toolbar>
        <Grid container spacing={2} alignItems="center">
          <Grid item>
            <SearchIcon color="inherit" className={classes.block} />
          </Grid>
          <Grid item xs>
            <TextField
              placeholder={placeholder}
              InputProps={{
                disableUnderline: true,
                className: classes.searchInput,
              }}
              fullWidth
            />
          </Grid>
          <Grid item>
            <Tooltip title="Refrescar datos">
              <IconButton onClick={handleRefresh}>
                <RefreshIcon color="inherit" className={classes.block} />
              </IconButton>
            </Tooltip>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

export default withStyles(displayData)(DisplayDataHeader);
