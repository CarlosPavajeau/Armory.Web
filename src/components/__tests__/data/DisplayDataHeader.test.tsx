import { render } from '@testing-library/react';
import DisplayDataHeader from '../../data/DisplayDataHeader';

describe('DisplayDataHeader tests', () => {
  test('Render display data header', () => {
    const placeholder = 'Test placeholder';
    let refresh = false;

    const handleRefresh = () => {
      refresh = !refresh;
    };

    const result = render(
      <DisplayDataHeader
        placeholder={placeholder}
        handleRefresh={handleRefresh}
      />,
    );
    expect(result).toBeTruthy();
    expect(result.getByPlaceholderText(placeholder)).toBeTruthy();
    expect(result.getByTitle('Refrescar datos')).toBeTruthy();

    const button = result.getByTitle('Refrescar datos') as HTMLButtonElement;
    button.click();

    expect(refresh).toEqual(true);
  });
});
