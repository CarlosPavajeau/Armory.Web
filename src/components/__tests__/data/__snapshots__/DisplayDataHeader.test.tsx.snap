// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayDataHeader tests Should be render DisplayDataHeader 1`] = `
<DisplayDataHeader
  classes={
    Object {
      "block": "DisplayDataHeader-block-4",
      "container": "DisplayDataHeader-container-5",
      "paper": "DisplayDataHeader-paper-1",
      "searchBar": "DisplayDataHeader-searchBar-2",
      "searchInput": "DisplayDataHeader-searchInput-3",
      "withoutData": "DisplayDataHeader-withoutData-6",
    }
  }
  handleRefresh={[Function]}
  placeholder="Test placeholder"
/>
`;
